File: Concurrent_Toy_Language_Interpreter/ADTs/MyTuple.java
package ADTs;

public class MyTuple<T1,T2> implements ITuple<T1,T2> {
    private T1 first;
    private T2 second;

    public MyTuple(T1 f, T2 s) {
        this.first = f;
        this.second = s;
    }

    public T1 getFirst() {
        return this.first;
    }

    public T2 getSecond() {
        return this.second;
    }

    public String toString() {
        return "(" + this.first.toString() + "," + this.second.toString() + ")";
    }
}

File: Toy_Language_Interpreter_C/ADTs/MyTuple.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace a7
{
    public class MyTuple<T1,T2>
    {
        private T1 first;
        private T2 second;

        public T1 First
        {
            get
            {
                return this.first;
            }

            set
            {
                this.first = value;
            }
        }

        public T2 Second
        {
            get
            {
                return this.second;
            }

            set
            {
                this.second = value;
            }
        }

        public MyTuple(T1 f, T2 s)
        {
            this.first = f;
            this.second = s;
        }

        public override string ToString()
        {
            return "(" + this.first.ToString() + "," + this.second.ToString() + ")";
        }
    }
}
